docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+
+- **"ML.NET learning and customization" scenario: (Previews)** .NET developers are able to research the generated pipeline's implementation code (C# + ML.NET API) for customization of the generated ML.NET code.
+
+- **"Cross-platform" scenario: (Previews)** .NET developers are able to use the ML.NET CLI from the same OS supported by .NET Core: Windows, Mac and Linux, so they are not tight to a single environment like Windows and VisualStudio but can be used from any OS CLI and code editor.
+
+- **"Advanced ML learning and customization" scenario: (v0.2)** Developers with ML skills are able to customize internal ML parameters such as trainerexclusion/inclusion lists, specific featurization, metrics to prioritize, etc. for more advanced learning and customization of the generated models and ML.NET code and even to improve total execution times by providing certain prescriptive ML "shortcuts".
+
+- **Azure integration for scaling-out (v1.0 or vNext)**: In next versions, an important goal will be to be able to scale-out the explorations and training in parallel tasks deployed into Azure in addition to running it on a local PC. AutoML in Azure is a great example, but regular training of a model could also be performed in Azure.
+
+The goals for the features is to automate the following steps when building a model to achieve the above business goals:
+
+**Foundational features:** 
+
+- Provide an end-to-end **ML.NET CLI** for developers (i.e. *"mlnet new"*) to generate either the final trained model and the pipeline's C#/ML.NET implementation code in a similar fashion to the [.NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/?tabs=netcore2x). The CLI is also a foundation upon which higher-level tools, such as Integrated Development Environments (IDEs) can rest.
+- Automatic selection of best columns (input variables) to be used by the model
+- Automatic featurization and feature engineering 
+- Automatic learner sweeping and selection
+- Automatic hyperparameter sweeping and selection
+  
+
+# Solution
+
+The solution that can target any kind of developer and fits especially well when generating code is a CLI (Command-Line Interface). 
+
+The main support reasons are:
+
+- A CLI can be the foundation where other higher level tooling (UI, IDEs) can also rest on.
+- A CLI makes sense as a tool for "code generation". For the end-user experience, just using a C# API to generate C# code would not make sense and would feel a bit weird for the developer experience used to .NET approaches.
+- A CLI is aligned to modern cross-platform frameworks experience such as .NET Core, Python, Node, etc. 
+
+## Positioning the CLI and AutoML in ML.NET
+
+![alt text](Images/MLNET-AutoML-Positioning.png "ML.NET and AutoML")
+
+As mentioned at the begining of the spec doc, the CLI will be branded as the ML.NET CLI since this CLI will also have additional features where AutoML is not needed.
```

Comment:
I'm not sure I follow what this sentence is trying to convey. 

`the CLI will be branded as the ML.NET CLI since this CLI will also have additional features where AutoML is not needed`.

"AutoML" is just a component of ML.NET.... This would be like saying "The .NET CLI will be branded as the .NET CLI since this CLI will also have additional features where Roslyn is not needed".